crypto/merkleTree.go

```diff
@@ -109,3 +110,31 @@ func ComputeRoot(hashes []common.Uint256) (common.Uint256, error) {
 	tree, _ := NewMerkleTree(hashes)
 	return tree.Root.Hash, nil
 }
+
+func VerifyRoot(txnsHash []common.Uint256, txnsRoot []byte) error {
+	if hasDuplicateHash(txnsHash) {
+		return fmt.Errorf("transactions contain duplicate hash")
+	}
+
+	computedRoot, err := ComputeRoot(txnsHash)
+	if err != nil {
+		return err
+	}
+
+	if !bytes.Equal(computedRoot.ToArray(), txnsRoot) {
+		return fmt.Errorf("computed txn root %x is different from txn root in header %x", computedRoot.ToArray(), txnsRoot)
+	}
+
+	return nil
+}
+
+func hasDuplicateHash(txnsHash []common.Uint256) bool {
+	count := make(map[common.Uint256]int, len(txnsHash))
+	for _, txnHash := range txnsHash {
+		if count[txnHash] > 0 {
```

Comment:
why not using 
`if _, ok := txnsHash[txnHash]; ok {
   return false
}
`
to check if hash exist?